cmd/kubeadm/app/apis/kubeadm/v1beta3/types.go

```diff
@@ -34,9 +34,11 @@ type InitConfiguration struct {
 
 	// BootstrapTokens is respected at `kubeadm init` time and describes a set of Bootstrap Tokens to create.
 	// This information IS NOT uploaded to the kubeadm cluster configmap, partly because of its sensitive nature
+	// +optional
```

Comment:
@rosti - could you outline the use case in the comment on why certain fields are optional, b/c the story is different and it's not clear to an observer what is defaulted and what is an option override.

Reply:
Sounds good. I'll add a short comment just above each optional marking.